Sources/StreamVideoSwiftUI/CallView/VideoRenderer.swift

```diff
@@ -112,6 +120,40 @@ public class VideoRenderer: RTCMTLVideoView {
         }
     }
     
+    public override func renderFrame(_ frame: RTCVideoFrame?) {
+        super.renderFrame(frame)
+        
+        guard let feedFrames else { return }
+        
+        skip.toggle()
+        if skip {
+           return
+        }
+        
+        DispatchQueue.global(qos: .userInitiated).async {
+            guard let frame = frame else {
+                return
+            }
+
+            if let pixelBuffer = frame.buffer as? RTCCVPixelBuffer {
+                guard let sampleBuffer = CMSampleBuffer.from(pixelBuffer.pixelBuffer) else {
+                    log.warning("Failed to convert CVPixelBuffer to CMSampleBuffer")
+                    return
+                }
+
+                feedFrames(sampleBuffer)
+            } else if let i420buffer = frame.buffer as? RTCI420Buffer {
+                let reductionFactor = UIDevice.current.userInterfaceIdiom == .pad ? 4 : 6
```

Comment:
Performance reasons again - we need to reduce the track resolution, since it's displayed in a smaller place.
The logic for picking them:
- roughly how much the PiP view takes in an average iPhone or iPad
- testing and tweaking based on CPU usage
- video quality tradeoff

Reply:
A comment explaining those details will be helpful.

Reply:
Good point, will add this 👍